react/week-21/lesson.md

```diff
@@ -327,100 +327,114 @@ render() {
       </div>
     );
   }
 }
 ```
 
-The key method here is `setInputRef`. It is called by React when rendering the `<input>`, and passes a reference to the real DOM node as an argument. We remember the reference by assigning it to `this.input`. Then when the button is clicked we can call the `focus` method (a vanilla method, not part of React) on the input DOM node.
-
-### Uncontrolled & Controlled Components
-
-Refs are also useful when building forms to collect user data. We can use them in a *pattern* called an *uncontrolled component*. A pattern is a repeated solution to a problem that is useful in multiple similar cases.
+We're controlling the `value` of the input by using the value from the `reminder` state. This means that we can only change the value by updating the state. It is done using the `onChange` attribute and the method `handleChange` which is called every time the input value changes (typically when a new character is added or removed). If you didn't call `this.setState()` in the `handleChange` method, then the input's value would never change and it would appear as if you couldn't type in the input! Finally, the value we keep in the `reminder` state is displayed on the screen as today's reminder. 
 
-Let's look an example of an uncontrolled component ([interactive example](https://codesandbox.io/s/04x2r6ko0p)):
-
-```js
-class UncontrolledComponent extends Component {
-  setInputRef = (inputEl) => {
-    this.inputRef = inputEl;
-  };
+In addition, instead of just saving the value of the input in the state, we could have also transformed the string before we set it with `this.setState()`, for example by calling `toUpperCase()` on the string.
 
-  handleSubmit = (event) => {
-    console.log(this.inputRef.value);
-    event.preventDefault(); // Prevents form submission
-  };
+Let's have a look at a more complex example where we want to build a form to let users enter information to create an personal account ([interactive example](https://codesandbox.io/s/m7p083zn6p)):
 
-  render() {
-    return (
-      <form onSubmit={this.handleSubmit}>
-        <input type="text" ref={this.setInputRef} placeholder="Name" />
-        <button type="submit">Submit</button>
-      </form>
-    );
-  }
-}
 ```
-
-By using a ref, we can gather all of the input data in the form at once and do something with it, for example send it in a POST request.
-
-In contrast, we can get more control over our input data by using the *controlled component* pattern. Let's look at an example ([interactive example](https://codesandbox.io/s/4jq1yqy8kx)):
-
-```js
-class ControlledComponent extends Component {
+class CreateAccountForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      name: ''
+      username: "",
+      email: "",
+      password: ""
     };
   }
 
-  handleChange = (event) => {
+  handleChange = event => {
     this.setState({
-      name: event.target.value
+      [event.target.name]: event.target.value
```

Comment:
Do you think it's worth commenting on this below? I suspect students will have forgotten about this (do we even teach it at all?)

Reply:
Yeah you're right, probably good to explain a bit this syntax